eventuate-common-in-memory-database-micronaut/src/main/java/io/eventuate/common/inmemorydatabase/EventuateCommonInMemoryDatabaseFactory.java

```diff
@@ -0,0 +1,23 @@
+package io.eventuate.common.inmemorydatabase;
+
+import io.micronaut.context.annotation.Factory;
+import org.springframework.jdbc.datasource.embedded.EmbeddedDatabaseBuilder;
+import org.springframework.jdbc.datasource.embedded.EmbeddedDatabaseType;
+
+import javax.inject.Singleton;
+import javax.sql.DataSource;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Factory
+public class EventuateCommonInMemoryDatabaseFactory {
+
+  @Singleton
+  public DataSource dataSource(List<EventuateDatabaseScriptSupplier> scripts) {
+    EmbeddedDatabaseBuilder builder = new EmbeddedDatabaseBuilder();
```

Comment:
Lines 17-20  are identical to the code in the Spring version.
Why duplicate? Extract and move into common-in-memory